flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/BigQuerySchemaProviderImpl.java

```diff
@@ -0,0 +1,347 @@
+        AVRO_TYPES_TO_PROTO.put(Schema.Type.STRING, FieldDescriptorProto.Type.TYPE_STRING);
+        AVRO_TYPES_TO_PROTO.put(Schema.Type.BOOLEAN, FieldDescriptorProto.Type.TYPE_BOOL);
+        AVRO_TYPES_TO_PROTO.put(Schema.Type.ENUM, FieldDescriptorProto.Type.TYPE_STRING);
+        AVRO_TYPES_TO_PROTO.put(Schema.Type.BYTES, FieldDescriptorProto.Type.TYPE_BYTES);
+
+        /*
+         * Map Logical Avro Schema Type to FieldDescriptorProto Type, which converts
+         * AvroSchema Primitive Type to Dynamic Message.
+         * LOGICAL_AVRO_TYPES_TO_PROTO: Map containing mapping from Primitive Avro Schema Type to FieldDescriptorProto.
+         */
+        LOGICAL_AVRO_TYPES_TO_PROTO = new HashMap<>();
+        LOGICAL_AVRO_TYPES_TO_PROTO.put(
+                LogicalTypes.date().getName(), FieldDescriptorProto.Type.TYPE_INT32);
+        LOGICAL_AVRO_TYPES_TO_PROTO.put(
+                LogicalTypes.decimal(1).getName(), FieldDescriptorProto.Type.TYPE_BYTES);
+        LOGICAL_AVRO_TYPES_TO_PROTO.put(
+                LogicalTypes.timestampMicros().getName(), FieldDescriptorProto.Type.TYPE_INT64);
+        LOGICAL_AVRO_TYPES_TO_PROTO.put(
+                LogicalTypes.timestampMillis().getName(), FieldDescriptorProto.Type.TYPE_INT64);
+        LOGICAL_AVRO_TYPES_TO_PROTO.put(
+                LogicalTypes.uuid().getName(), FieldDescriptorProto.Type.TYPE_STRING);
+        // These are newly added.
+        LOGICAL_AVRO_TYPES_TO_PROTO.put(
+                LogicalTypes.timeMillis().getName(), FieldDescriptorProto.Type.TYPE_STRING);
+        LOGICAL_AVRO_TYPES_TO_PROTO.put(
+                LogicalTypes.timeMicros().getName(), FieldDescriptorProto.Type.TYPE_STRING);
+        LOGICAL_AVRO_TYPES_TO_PROTO.put(
+                LogicalTypes.localTimestampMillis().getName(),
+                FieldDescriptorProto.Type.TYPE_STRING);
+        LOGICAL_AVRO_TYPES_TO_PROTO.put(
+                LogicalTypes.localTimestampMicros().getName(),
+                FieldDescriptorProto.Type.TYPE_STRING);
+        LOGICAL_AVRO_TYPES_TO_PROTO.put("geography_wkt", FieldDescriptorProto.Type.TYPE_STRING);
+        LOGICAL_AVRO_TYPES_TO_PROTO.put("Json", FieldDescriptorProto.Type.TYPE_STRING);
+    }
+
+    // --------------- Obtain AvroSchema from TableSchema -----------------
+    /**
+     * Function to convert TableSchema to Avro Schema.
+     *
+     * @param tableSchema A {@link TableSchema} object to cast to {@link Schema}.
+     * @return Converted Avro Schema
+     */
+    private static Schema getAvroSchema(TableSchema tableSchema) {
+        return SchemaTransform.toGenericAvroSchema("root", tableSchema.getFields());
+    }
+
+    // --------------- Obtain Descriptor Proto from Avro Schema  ---------------
+    /**
+     * Obtains a Descriptor Proto by obtaining Descriptor Proto field by field.
+     *
+     * <p>Iterates over Avro Schema to obtain FieldDescriptorProto for it.
```

Comment:
nit: "to obtain the FieldDescriptorProto"

Reply:
Fixed.

Reply:
thanks! you can resolve this convo